accounts-db/src/tiered_storage/readable.rs

```diff
@@ -73,11 +76,16 @@ impl<'accounts_file, M: TieredAccountMeta> ReadableAccount
 
     /// Returns the epoch that this account will next owe rent by parsing
     /// the specified account block.  Epoch::MAX will be returned if the account
-    /// is rent-exempt.
+    /// is rent-exempt, and 0 for any zero-lamport account.
     fn rent_epoch(&self) -> Epoch {
         self.meta
             .rent_epoch(self.account_block)
-            .unwrap_or(Epoch::MAX)
+            .unwrap_or(if self.lamports() != 0 {
```

Comment:
I observed this when HotStorage fails `accounts_db::tests::test_clean_zero_lamport_and_dead_slot` test (or some other tests with zero_lamport) so not sure whether it is checked somewhere in the code.  Btw, is rent-epoch part of any hash calculation?  If the answer is yes, then this will introduce different behavior between the current AppendVec accounts and HotStorage accounts.

Reply:
Thanks, @brooksprumo!  I think this explains why HotStorage previously get bank hash mismatch as it uses a different rent-epoch for zero-lamport accounts (AppendVec uses 0 while HotStorage uses `RENT_EXEMPT_RENT_EPOCH`.)

To move forward, I think we have two options here:

* Keep the existing behavior untouched at least until we can fully migrate to HotStorage.  Use the current PR as a temporary approach to keep the AppendVec code path untouched.

* In the hash calculation, convert RENT_EXEMPT_RENT_EPOCH to 0 when `lamports` is zero -- but I am not sure whether there're any case where AppendVec uses some other value and might later cause hash mis-match between different versions.

Reply:
Sounds good.  Will improve the comment.

Reply:
Updated.  Let me know if it looks good to you.